vulnerabilities/pipelines/compute_package_risk.py

```diff
@@ -26,17 +28,57 @@ class ComputePackageRiskPipeline(VulnerableCodePipeline):
 
     @classmethod
     def steps(cls):
-        return (cls.add_package_risk_score,)
+        return (
+            cls.compute_and_store_vulnerability_risk_score,
+            cls.compute_and_store_package_risk_score,
+        )
+
+    def compute_and_store_vulnerability_risk_score(self):
+        affected_vulnerabilities = Vulnerability.objects.filter(
+            affectedbypackagerelatedvulnerability__isnull=False
+        ).prefetch_related(
+            "references",
+            "exploits",
+        )
 
-    def add_package_risk_score(self):
+        self.log(
+            f"Calculating risk for {affected_vulnerabilities.count():,d} vulnerability with a affected packages records"
+        )
+
+        progress = LoopProgress(total_iterations=affected_vulnerabilities.count(), logger=self.log)
+
+        updatables = []
+        updated_vulnerability_count = 0
+        batch_size = 5000
+
+        for vulnerability in progress.iter(affected_vulnerabilities.paginated()):
+            severities = vulnerability.severities.all()
+            references = vulnerability.references.all()
+
+            (
+                vulnerability.weighted_severity,
+                vulnerability.exploitability,
+            ) = compute_vulnerability_risk_factors(references, severities, vulnerability.exploits)
+
+            updatables.append(vulnerability)
+
+            if len(updatables) >= batch_size:
+                updated_vulnerability_count += bulk_update_vulnerability_risk_score(
+                    vulnerabilities=updatables,
+                    logger=self.log,
+                )
+        updated_vulnerability_count += bulk_update_vulnerability_risk_score(
+            vulnerabilities=updatables,
+            logger=self.log,
+        )
+        self.log(
+            f"Successfully added risk score for {updated_vulnerability_count:,d} vulnerability"
+        )
+
+    def compute_and_store_package_risk_score(self):
         affected_packages = (
-            Package.objects.filter(affected_by_vulnerabilities__isnull=False).prefetch_related(
```

Comment:
Seems to me Keshav changes are removed

Reply:
@TG1999, I've updated the code, but Keshav left some comments. I'll update it now